packages/runtime/src/utils/share.ts

```diff
@@ -173,6 +178,9 @@ export function getRegisteredShare(
           sc,
           pkgName,
         );
+        if (typeof singletonVersion === 'function') {
```

Comment:
if its a function. not string, then user must do all the work to return a valid module from federation. if its string we do like normal load strategy